File: Dummy/src/Frontend/next/Web/src/App/src/lib/DomainUseCases/App/Actions/Login/AppLoginActionCommand.ts
export interface AppLoginActionCommand {
  readonly userName: string;
  readonly password: string;
}

export function createAppLoginActionCommand(
  options?: Partial<AppLoginActionCommand> | null
): AppLoginActionCommand {
  return {
    userName: options?.userName ?? '',
    password: options?.password ?? '',
  };
}

File: Dummy/src/Backend/src/Gateway/src/DomainUseCases/App/Actions/Login/AppLoginActionCommand.cs
namespace Makc2025.Dummy.Gateway.DomainUseCases.App.Actions.Login;

/// <summary>
/// Команда действия по входу в приложение.
/// </summary>
/// <param name="UserName">Имя пользователя.</param>
/// <param name="Password">Пароль.</param>
public record AppLoginActionCommand(
  string UserName,
  string Password) : ICommand<Result<AppLoginActionDTO>>;
